controllers/common/ansible.go

```diff
@@ -5,25 +5,109 @@ package common
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/dynamic"
+	ctrl "sigs.k8s.io/controller-runtime"
 
 	policyv1beta1 "open-cluster-management.io/governance-policy-propagator/api/v1beta1"
 )
 
+const (
+	ControllerName             string = "policy-automation"
+	PolicyAutomationLabel      string = "ansiblejob.policy.open-cluster-management.io/policyautomation-name"
+	PolicyAutomationGeneration string = "policy.open-cluster-management.io/policyautomation-generation"
+	AnsiblejobIndex            string = "policy.open-cluster-management.io/ansiblejob-index"
+)
+
+var log = ctrl.Log.WithName(ControllerName)
+
+var ansibleJobRes = schema.GroupVersionResource{
+	Group: "tower.ansible.com", Version: "v1alpha1",
+	Resource: "ansiblejobs",
+}
+
+// Check ansiblejob exist or not and if exist returns ansiblejob
+func GetAnsibleJobOrder(policyAutomation *policyv1beta1.PolicyAutomation,
+	dynamicClient dynamic.Interface,
+) int {
+	ansiblejobList, err := dynamicClient.Resource(ansibleJobRes).
+		Namespace(policyAutomation.GetNamespace()).List(
+		context.TODO(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", PolicyAutomationLabel, policyAutomation.GetName()),
+		},
+	)
+	if err != nil {
+		return 0
+	}
+
+	return len(ansiblejobList.Items) + 1
+}
+
+// Check ansiblejob already exist
+func GetLastAjGeneration(policyAutomation *policyv1beta1.PolicyAutomation,
+	dynamicClient dynamic.Interface,
+) (string, error) {
+	ansiblejobList, err := dynamicClient.Resource(ansibleJobRes).Namespace(policyAutomation.GetNamespace()).List(
+		context.TODO(), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", PolicyAutomationLabel, policyAutomation.GetName()),
+		},
+	)
+
+	ansiblejobLen := len(ansiblejobList.Items)
+
+	if err != nil {
+		log.Error(err, "Faile to get ansiblejob list")
+
+		return "0", err
+	}
+
+	if ansiblejobLen == 0 {
+		return "0", nil
+	}
+
+	// sort by AnsiblejobIndex annotation
+	sort.Slice(ansiblejobList.Items, func(i, j int) bool {
+		p1, err := strconv.Atoi(ansiblejobList.Items[i].GetAnnotations()[AnsiblejobIndex])
+		if err != nil {
+			log.Error(err, "Failed to sort ansiblejob by index annotation at the first")
+		}
+		p2, err := strconv.Atoi(ansiblejobList.Items[j].GetAnnotations()[AnsiblejobIndex])
+		if err != nil {
+			log.Error(err, "Failed to sort ansiblejob by index annotation at the second")
+		}
+
+		return p1 < p2
+	})
+	lastAJAnnotation := ansiblejobList.Items[ansiblejobLen-1].GetAnnotations()
+	stringGeneration := lastAJAnnotation[PolicyAutomationGeneration]
+
+	return stringGeneration, nil
+}
+
 // CreateAnsibleJob creates ansiblejob with given PolicyAutomation
 func CreateAnsibleJob(policyAutomation *policyv1beta1.PolicyAutomation,
 	dynamicClient dynamic.Interface, mode string, violationContext policyv1beta1.ViolationContext,
 ) error {
+	order := GetAnsibleJobOrder(policyAutomation, dynamicClient)
```

Comment:
It seems like the console might already sort it based on start and finish time, so that might be good enough? https://github.com/stolostron/console/blob/main/frontend/src/routes/Governance/components/AutomationDetailsSidebar.tsx#L102 (note, this is just the first place I found, I didn't check that it's the table you might be thinking of)

Reply:
I should get the latest ansiblejob.  I used timestamp for sort, but the smallest unit is second so hard to sort.